src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/ListenPortLoader.java

```diff
@@ -0,0 +1,72 @@
+import org.w3c.dom.Element;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.ListenPort.ClientAuthentication.fromType;
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.BuilderUtils.mapPropertiesElements;
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.BundleElementNames.*;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentTools.*;
+import static java.lang.Integer.parseInt;
+
+public class ListenPortLoader implements BundleEntityLoader {
+
+    ListenPortLoader() {
+    }
+
+    @Override
+    public void load(final Bundle bundle, final Element element) {
+        final Element listenPortElement = getSingleChildElement(getSingleChildElement(element, RESOURCE), LISTEN_PORT);
+
+        final String name = getSingleChildElement(listenPortElement, NAME).getTextContent();
+        final String protocol = getSingleChildElement(listenPortElement, PROTOCOL).getTextContent();
+        final Integer port = parseInt(getSingleChildElement(listenPortElement, PORT).getTextContent());
+        final List<String> enabledFeatures = getChildElementsTextContents(getSingleChildElement(listenPortElement, ENABLED_FEATURES), STRING_VALUE);
+        final String targetServiceReference = getTargetServiceReference(listenPortElement);
+        final ListenPortTlsSettings tlsSettings = getTlsSettings(listenPortElement);
+        final Map<String, Object> properties = mapPropertiesElements(getSingleChildElement(listenPortElement, PROPERTIES, true));
+
+        ListenPort listenPort = new ListenPort();
+        listenPort.setProtocol(protocol);
+        listenPort.setPort(port);
+        listenPort.setEnabledFeatures(enabledFeatures);
+        listenPort.setTargetServiceReference(targetServiceReference);
+        listenPort.setTlsSettings(tlsSettings);
+        listenPort.setProperties(properties);
+
+        bundle.getListenPorts().put(name, listenPort);
+    }
+
+    private String getTargetServiceReference(final Element listenPortElement) {
+        Element targetServiceReference = getSingleChildElement(listenPortElement, TARGET_SERVICE_REFERENCE);
+        if (targetServiceReference == null) {
+            return null;
+        }
+        return targetServiceReference.getTextContent();
+    }
+
+    private ListenPortTlsSettings getTlsSettings(final Element listenPortElement) {
+        Element tlsSettingsElement = getSingleChildElement(listenPortElement, TLS_SETTINGS);
+        if (tlsSettingsElement == null) {
+            return null;
+        }
+
+        ListenPortTlsSettings tlsSettings = new ListenPortTlsSettings();
+        tlsSettings.setClientAuthentication(fromType(getSingleChildElementTextContent(tlsSettingsElement, CLIENT_AUTHENTICATION)));
+        tlsSettings.setEnabledCipherSuites(getChildElementsTextContents(getSingleChildElement(tlsSettingsElement, ENABLED_VERSIONS, true), STRING_VALUE));
```

Comment:
I would consider separating some of these lines into multiple lines for better readability